bazel/api-gen/extraction/extract_api_to_json.bzl

```diff
@@ -19,6 +19,12 @@ def _extract_api_to_json(ctx):
     json_output = ctx.outputs.output_name
     args.add(json_output.path)
 
+    # Pass the import path map
+    path_map = {}
+    for target, path in ctx.attr.import_map.items():
+        path_map[path] = target.files.to_list()[0].path
```

Comment:
I recommend checking via `len` if there are more files and throwing then (using `fail`)